trie/database.go

```diff
@@ -333,35 +347,49 @@ func (db *Database) insertPreimage(hash common.Hash, preimage []byte) {
 // node retrieves a cached trie node from memory, or returns nil if none can be
 // found in the memory cache.
 func (db *Database) node(hash common.Hash, cachegen uint16) node {
-	// Retrieve the node from cache if available
+	// Retrieve the node from the clean cache if available
+	if enc, err := db.cleans.Get(string(hash[:])); err == nil || enc != nil {
+		return mustDecodeNode(hash[:], enc, cachegen)
+	}
+	// Retrieve the node from the dirty cache if available
 	db.lock.RLock()
-	node := db.nodes[hash]
+	dirty := db.nodes[hash]
```

Comment:
`dirty` here simply means not-yet-written-to-disk, right?

Reply:
And if so, maybe we should rename `db.nodes` into `db.nodesNotOnDisk` ... or maybe not, just a thought